src/main/java/org/jfrog/hudson/pipeline/declarative/steps/BuildAppendStep.java

```diff
@@ -0,0 +1,92 @@
+package org.jfrog.hudson.pipeline.declarative.steps;
+
+import com.google.inject.Inject;
+import hudson.Extension;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jfrog.hudson.pipeline.ArtifactorySynchronousStepExecution;
+import org.jfrog.hudson.pipeline.common.executors.BuildAppendExecutor;
+import org.jfrog.hudson.pipeline.common.types.ArtifactoryServer;
+import org.jfrog.hudson.pipeline.common.types.buildInfo.BuildInfo;
+import org.jfrog.hudson.pipeline.declarative.utils.DeclarativePipelineUtils;
+import org.jfrog.hudson.util.JenkinsBuildInfoLog;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import java.io.IOException;
+
+@SuppressWarnings("unused")
+public class BuildAppendStep extends AbstractStepImpl {
+
+    public static final String STEP_NAME = "rtBuildAppend";
+    private final String buildNumberToAppend;
+    private final String buildNameToAppend;
+    private final String serverId;
+    private String buildNumber;
+    private String buildName;
+
+    @DataBoundConstructor
+    public BuildAppendStep(String serverId, String buildNameToAppend, String buildNumberToAppend) {
+        this.serverId = serverId;
+        this.buildNameToAppend = buildNameToAppend;
+        this.buildNumberToAppend = buildNumberToAppend;
+    }
+
+    @DataBoundSetter
+    public void setBuildName(String buildName) {
+        this.buildName = buildName;
+    }
+
+    @DataBoundSetter
+    public void setBuildNumber(String buildNumber) {
+        this.buildNumber = buildNumber;
+    }
+
+    public static class Execution extends ArtifactorySynchronousStepExecution<Void> {
+
+        private final transient BuildAppendStep step;
+
+        @Inject
+        public Execution(BuildAppendStep step, StepContext context) throws IOException, InterruptedException {
+            super(context);
+            this.step = step;
+        }
+
+        @Override
+        protected Void run() throws Exception {
+            ArtifactoryServer server = DeclarativePipelineUtils.getArtifactoryServer(build, ws, getContext(), step.serverId);
+            BuildInfo buildInfo = DeclarativePipelineUtils.getBuildInfo(ws, build, step.buildName, step.buildNumber);
+            if (buildInfo == null) {
+                throw new RuntimeException("Build " + DeclarativePipelineUtils.createBuildInfoId(build, step.buildName, step.buildNumber) + " does not exist!");
```

Comment:
Let's also add the Artifactory URL to this exception message. I think it'll help during troubleshooting, when trying to figure out where wasn't the build found at.

Reply:
The getBuildInfo extract the information from the file system. It doesn't download anything from Artifactory.